plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/IcebergMetadata.java

```diff
@@ -1729,11 +1730,30 @@ private void dropField(ConnectorSession session, ConnectorTableHandle tableHandl
     @Override
     public void renameColumn(ConnectorSession session, ConnectorTableHandle tableHandle, ColumnHandle source, String target)
     {
+        Table icebergTable = catalog.loadTable(session, ((IcebergTableHandle) tableHandle).getSchemaTableName());
         IcebergColumnHandle columnHandle = (IcebergColumnHandle) source;
+        renameColumn(icebergTable, columnHandle.getName(), target);
+    }
+
+    @Override
+    public void renameField(ConnectorSession session, ConnectorTableHandle tableHandle, List<String> fieldPath, String target)
+    {
         Table icebergTable = catalog.loadTable(session, ((IcebergTableHandle) tableHandle).getSchemaTableName());
+        String parentPath = String.join(".", fieldPath.subList(0, fieldPath.size() - 1));
+        NestedField parent = icebergTable.schema().caseInsensitiveFindField(parentPath);
+
+        String caseSensitiveParentName = icebergTable.schema().findColumnName(parent.fieldId());
+        NestedField source = parent.type().asStructType().caseInsensitiveField(getLast(fieldPath));
+
+        String sourcePath = caseSensitiveParentName + "." + source.name();
+        renameColumn(icebergTable, sourcePath, target);
+    }
+
+    private void renameColumn(Table icebergTable, String source, String target)
```

Comment:
Looking at this method one need to think what `source` and `target` is (dotted strings)
This method is trivial enough, and used in different contexts, to warrant inlining.

After doing that, change "Failed to rename column" message for renamefield case to "Failed to rename field"